test/core/client_channel/lb_policy/pick_first_test.cc

```diff
@@ -177,6 +251,54 @@ TEST_F(PickFirstTest, GoesIdleWhenConnectionFailsThenCanReconnect) {
   }
 }
 
+TEST_F(PickFirstTest, WithShuffle) {
+  testing::ScopedExperimentalEnvVar env_var(
+      "GRPC_EXPERIMENTAL_PICKFIRST_LB_CONFIG");
+  // 6 addresses have 6! = 720 permutations or 0.1% chance that the shuffle
+  // returns initial sequence
+  constexpr std::array<absl::string_view, 6> kAddresses = {
+      "ipv4:127.0.0.1:443", "ipv4:127.0.0.1:444", "ipv4:127.0.0.1:445",
+      "ipv4:127.0.0.1:446", "ipv4:127.0.0.1:447", "ipv4:127.0.0.1:448"};
+  absl::Status status = ApplyUpdate(
```

Comment:
Now that we're going to do the address shuffling in `UpdateLocked()` instead of in `AttemptToConnectUsingLatestUpdateArgsLocked()`, I think we can just move this `ApplyUpdate()` call inside of `GetOrderAddressesArePicked()`, so that we trigger a new shuffle each time we call it.

Reply:
I did it slightly different so GetOrderAddressesArePicked can be used to verify addresses are not reshuffled if all subchannels fail